tests/test_job_management.py

```diff
@@ -2639,6 +2640,383 @@ def test_inspect_status(dispatcher_live_fixture, request_cred, roles, include_se
+    token_none = (request_cred == 'public')
+
+    token_payload = {**default_token_payload}
+
+    if token_none:
+        encoded_token = None
+    else:
+        # let's generate a valid token
+        token_payload["roles"] = roles
+        encoded_token = jwt.encode(token_payload, secret_key, algorithm='HS256')
+
+    params = {
+        'query_status': 'new',
+        'product_type': 'dummy',
+        'query_type': "Dummy",
+        'instrument': 'empty',
+        'token': encoded_token,
+    }
+
+    # just for having the roles in a list
+    roles = roles.split(',')
+    roles[:] = [r.strip() for r in roles]
+
+    jdata_done = ask(server,
+                params,
+                expected_query_status=["done"],
+                max_time_s=150,
+                )
+
+    params = {
+        'query_status': 'new',
+        'product_type': 'failing',
+        'query_type': "Dummy",
+        'instrument': 'empty',
+        'token': encoded_token,
+    }
+
+    jdata_failed = ask(server,
+                       params,
+                       expected_query_status='failed'
+                       )
+
+    job_id_done = jdata_done['job_monitor']['job_id']
+    session_id_done = jdata_done['session_id']
+    job_id_failed = jdata_failed['job_monitor']['job_id']
+    session_id_failed = jdata_failed['session_id']
+
+    # generate a new valid token with the same approach
+    if token_none:
+        encoded_token = None
+    else:
+        # let's generate a valid token
+        token_payload["roles"] = roles
+        token_payload["exp"] = int(time.time()) + 5000
+        encoded_token = jwt.encode(token_payload, secret_key, algorithm='HS256')
+
+    status_code = 403
+    error_message = ''
+    if request_cred == 'invalid_token':
+        # an invalid (encoded) token, just a string
+        encoded_token = 'invalid_token'
+        error_message = 'The token provided is not valid.'
+    elif request_cred == 'public':
+        error_message = 'A token must be provided.'
+    elif request_cred == 'private':
+        if 'job manager' not in roles:
+            lacking_roles = ", ".join(sorted(list(set(required_roles) - set(roles))))
+            error_message = (
+                f'Unfortunately, your privileges are not sufficient to inspect the state for a given job_id.\n'
+                f'Your privilege roles include {roles}, but the following roles are missing: {lacking_roles}.'
+            )
+
+    inspect_params = dict(
+                         token=encoded_token
+                     )
+    if pass_job_id:
+        inspect_params['job_id'] = job_id_done[:8]
+
+    c = requests.get(server + "/inspect-jobs",
+                     params=inspect_params)
+
+    if request_cred != 'private' or ('job manager' not in roles):
+        # email not supposed to be sent for public request
+        assert c.status_code == status_code
+        assert c.text == error_message
+    else:
+        jdata= c.json()
+        assert 'jobs' in jdata
```

Comment:
that's a lot of asserts! Could you maybe use a json schema instead, or something like that?

Reply:
json schema is there, and the in the tests the output will be validated against it